### PR TITLE
Bulk recording metadata lookup

### PR DESCRIPTION
**Pull request by amCap1712**
POST endpoint for looking up multiple recording mbids at once, useful in troi for resolving mbids for imported playlist data.

### CODE REVIEW

File: listenbrainz/webserver/views/metadata_api.py [hunk 1]
```diff
@@ -295,6 +297,137 @@ def get_mbid_mapping():
     return jsonify({})
 
 
+def make_acrr_input(recording):
+    if "release_name" in recording:
```
Comment (mayhem) [inline listenbrainz/webserver/views/metadata_api.py:301]: docstrings

File: listenbrainz/webserver/views/metadata_api.py [hunk 2]
```diff
@@ -295,6 +297,137 @@ def get_mbid_mapping():
...
+    return all_results, all_params
+
+
+@metadata_bp.route("/lookup/", methods=["POST"])
+@crossdomain
+@ratelimit()
+def get_mbid_mapping_post():
+    """
+    This endpoint is the POST version for looking up recording mbids and associated MusicBrainz data. It allows up to
+    max number of items allowed. (:data:`~webserver.views.api.MAX_ITEMS_PER_GET` items)
```
Comment (mayhem) [inline listenbrainz/webserver/views/metadata_api.py:360]: I think for this endpoint we need to lower the max number of items per call. Sending us 1,000 items to look up and then doing so in quick succession is basically a DoS attack. :(

File: listenbrainz/webserver/views/metadata_api.py [hunk 3]
```diff
@@ -295,6 +297,137 @@ def get_mbid_mapping():
...
+
+        all_params[idx] = recording
+        all_results.append({
+            "artist_name_arg": artist_name,
+            "recording_name_arg": recording_name,
+            "release_name_arg": release_name,
+            "index": idx
+        })
+
+    try:
```
Comment (mayhem) [inline listenbrainz/webserver/views/metadata_api.py:415]: The scalability of this really scares me. :(

**Review by mayhem — APPROVED** [review]

File: listenbrainz/webserver/views/metadata_api.py (hunk 2 above)
Comment (amCap1712) [reply listenbrainz/webserver/views/metadata_api.py]: The aim is to support import of playlists using Troi, an alternative would be to have import logic in LB codebase instead but that would be less ideal. I have reduced the limit to 50 items per call.

File: listenbrainz/webserver/views/metadata_api.py (hunk 1 above)
Comment (amCap1712) [reply listenbrainz/webserver/views/metadata_api.py:304]: done

File: listenbrainz/webserver/views/metadata_api.py (hunk 3 above)
Comment (amCap1712) [reply listenbrainz/webserver/views/metadata_api.py:424]: True, we'll have to work out something better eventually.

**Pull request merged**